cloud/satellite.go

```diff
@@ -198,37 +198,60 @@ func (c *Client) ReserveSatellite(ctx context.Context, name, orgName, gitAuthor,
 		ctxTimeout, cancel := context.WithTimeout(ctx, 10*time.Minute)
 		defer cancel()
 		defer close(out)
-		stream, err := c.compute.ReserveSatellite(c.withAuth(ctxTimeout), &pb.ReserveSatelliteRequest{
-			OrgId:          orgID,
-			Name:           name,
-			CommitEmail:    gitAuthor,
-			GitConfigEmail: gitConfigEmail,
-			IsCi:           isCI,
-		})
-		if err != nil {
-			out <- SatelliteStatusUpdate{Err: errors.Wrap(err, "failed opening satellite reserve stream")}
-			return
-		}
-		for {
-			update, err := stream.Recv()
-			if err == io.EOF {
-				return
-			}
+		for i := 1; i <= 5; i++ {
```

Comment:
I see that there's another timeout above. But the hard-coded attempt count is less desirable than allowing the process to retry indefinitely within a max time window. Depending on the particular issue, the code may fail 5 times immediately or spend a while on each attempt. This library is pretty good and it can retry until a context times out: https://github.com/avast/retry-go

Reply:
I do like that library, but I'm also okay with the simple for-loop for now until we see if this is improving things or not. It may also be a temporary while we figure out what is going on with the unexpected hang-ups anyway.

Reply:
SGTM :+1: